ml-agents/mlagents/trainers/models_torch.py

```diff
@@ -142,6 +179,116 @@ def forward(self, vec_inputs, vis_inputs, memories=None, sequence_length=1):
         return embedding, memories
 
 
+class QNetwork(NetworkBody):
```

Comment:
The actor critic contains a NetworkBody, whereas this class inherits from one. Is there a reason for the difference? If there is a strong enough reason, maybe we should change ActorCritic to use the same pattern here?

Reply:
Or alternatively use the AC pattern here.